File: Python/is_prime.py
from math import ceil, sqrt


def is_prime(n: int) -> bool:
    """Improved Primality test"""
    if n < 3 or not n % 2:
        return n == 2
    for i in range(3, ceil(sqrt(n)) + 1, 2):
        if not n % i:
            return False
    return True

File: JavaScript/isPrime.js
// Imporoved Primality test //

function isPrime(x) {
    if (x < 3 || !(x % 2)) return x === 2
    for (let i = 3; i <= Math.sqrt(x); i += 2) 
        if (!(x % i)) 
            return false
    return true
}
